src/mixins/form-text.js

```diff
@@ -175,28 +151,20 @@ export default {
       if (evt.target.composing) {
         return
       }
-      const formatted = this.getFormatted(evt.target.value, evt)
+      const value = evt.target.value
+      const formattedValue = this.formatValue(value, evt)
       // Exit when the `formatter` function strictly returned `false`
       // or prevented the input event
       /* istanbul ignore next */
-      if (formatted === false || evt.defaultPrevented) {
+      if (formattedValue === false || evt.defaultPrevented) {
         evt.preventDefault()
         return
       }
-      this.updateValue(formatted, false)
-      this.$emit('change', formatted)
+      this.localValue = formattedValue
+      this.$emit('update', formattedValue)
+      this.$emit('change', formattedValue)
     },
     onBlur(evt) {
```

Comment:
Lazy formatting should be performed here on Blur, and not on change, as change can be triggered by character composition (for Chinese IME composition), mobile browser auto-complete or auto-correct.  Normally when blur happens a change event is fired first ,follewed by the blur event.